java-squid/src/main/java/org/sonar/java/resolve/Type.java

```diff
@@ -278,6 +278,10 @@ public Type erasure() {
       super(symbol);
       this.rawType = symbol.getType();
       this.typeSubstitution = typeSubstitution;
+      if (rawType.isTagged(UNKNOWN)) {
```

Comment:
I don't think this should be done at this level. I tend to think that when constructing a ParametrizedTypeType using the ParametrizedTypeCache we should return the unknown type altogether rather than an unknown ParametrizedTypeType which can (IMO) lead to confusion.